src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -1446,6 +1447,12 @@ StoreQueryResult(CitusScanState *scanState, MultiConnection *connection,
 
 			commandFailed = true;
 
+			/* an error happened, there is nothing we can do more */
+			if (fatalError)
```

Comment:
I don't think the boolean adds much, could just put `resultStatus == PGRES_FATAL_ERROR)` here